daemon/list.go

```diff
@@ -137,6 +138,14 @@ func (daemon *Daemon) Containers(config *ContainersConfig) ([]*types.Container,
 			Names: names[container.ID],
 		}
 		newC.Image = container.Config.Image
+		if container.Config.Image != "" {
+			img, err := daemon.repositories.LookupImage(container.Config.Image)
+			if err == nil {
+				if stringid.TruncateID(img.ID) == stringid.TruncateID(container.Config.Image) {
```

Comment:
I tried really hard to understand this change, but I just can't figure it out.
If the container's image's name == the image's ID (after truncating) then we set the result to the image ID.  But isn't it already set to the image ID?